hal/mbed_mpu_api.c

```diff
@@ -0,0 +1,120 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "hal/mpu_api.h"
+#include "cmsis.h"
+
+void mbed_mpu_init()
+{
+    mbed_mpu_enable_ram_xn(false);
+}
+
+void mbed_mpu_free()
+{
+    mbed_mpu_enable_ram_xn(false);
+}
+
+void mbed_mpu_enable_ram_xn(bool enable)
+{
+#if defined (__MPU_PRESENT) && (__MPU_PRESENT == 1U)
+    const uint32_t regions = (MPU->TYPE & MPU_TYPE_DREGION_Msk) >> MPU_TYPE_DREGION_Pos;
+
```

Comment:
I think we should have __DMB() here to ensure all outstanding memory transactions are completed before we start configuring the MPU.

Reply:
done